app/controllers/api/v0/habits_controller.rb

```diff
@@ -7,28 +7,47 @@ def index
   end
 
   def create
-    habit = @user.habits.create!(habit_params)
-    render json: HabitSerializer.new(habit), status: :created
+    habit = @user.habits.new(habit_params)
+    if habit.save
+      UpdateHabitStatusJob.set(wait_until: habit.end_datetime.tomorrow.beginning_of_day).perform_later(habit.id)
+      render json: HabitSerializer.new(habit), status: :created
+    else
+      render json: habit.errors.as_json(full_messages: true), status: 400
+    end
   end
 
-  def update
-    habit = @user.habits.find(params[:id])
-    habit.update(habit_params)
+  # def update
+  #   habit = @user.habits.find(params[:id])
+  #   habit.update(habit_params)
 
-    render json: HabitSerializer.new(habit), status: :accepted
-  end
+  #   render json: HabitSerializer.new(habit), status: :accepted
+  # end
 
   def destroy
     habit = @user.habits.find(params[:id])
-    habit.destroy
-    
-    render json: { message: "#{habit.name} has been deleted" }, status: :accepted
+    if habit.destroy
+      render json: { message: "#{habit.name} has been deleted" }, status: :accepted
+    else
+      render json: habit.errors.as_json(full_messages: true), status: 422
```

Comment:
I will merge these changes, but I want to include one more error handler that will cover all possible error scenarios: 

`rescue_from ActiveRecord::RecordNotDestroyed, with: :record_not_destroyed_response`

As the name implies, it will gracefully handle any exception related to the identified user not being deleted from the database.